backend/de.metas.shipper.gateway.nshift/src/main/java/de/metas/shipper/gateway/nshift/NShiftDraftDeliveryOrderCreator.java

```diff
@@ -216,13 +215,13 @@ private Optional<BigDecimal> computeNominalGrossWeightInKg(final PackageItem pac
 	public static PackageDimensions getPackageDimensions(@NonNull final PackageId packageId, @NonNull final UomId toUomId)
 	{
 		final IHUPackingMaterialDAO packingMaterialDAO = Services.get(IHUPackingMaterialDAO.class);
-		final I_M_HU_PackingMaterial packingMaterial = packingMaterialDAO.retrievePackingMaterialOrNull(packageId);
+		final PackageDimensions packageDimensions = packingMaterialDAO.retrievePackageDimensionsOrNull(packageId, toUomId);
```

Comment:
i would prefer a method that does not return null.
and usually, on service level i would prefer methods returning Optional than "OrNull"